core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -2088,6 +2088,9 @@ void rollbackOrProcessStateUpdates(
                     state.completeStateTransition(true);
                     // Cancel the acquisition lock timeout task for the state since it is acknowledged/released successfully.
                     state.cancelAndClearAcquisitionLockTimeoutTask();
+                    if (state.state != RecordState.ARCHIVED) {
+                        findNextFetchOffset.set(true);
+                    }
```

Comment:
This change here makes sense, but do we need to remove `findNextFetchOffset.set(true);` at other places where we just started the transaction i.e. in acknowledgement?